sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala

```diff
@@ -647,3 +647,29 @@ private[sql] class TestHiveSessionStateBuilder(
 
   override protected def newBuilder: NewBuilder = new TestHiveSessionStateBuilder(_, _)
 }
+
+private[hive] object HiveTestJars {
+  private val repository = "https://repository.apache.org/content/repositories/releases/"
+
+  def getHiveContribJar: File =
+    getJarFromUrl(s"${repository}org/apache/hive/hive-contrib/" +
+      s"${HiveUtils.builtinHiveVersion}/hive-contrib-${HiveUtils.builtinHiveVersion}.jar")
+  def getHiveHcatalogCoreJar: File =
+    getJarFromUrl(s"${repository}org/apache/hive/hcatalog/hive-hcatalog-core/" +
+      s"${HiveUtils.builtinHiveVersion}/hive-hcatalog-core-${HiveUtils.builtinHiveVersion}.jar")
+
+  private def getJarFromUrl(urlString: String): File = {
+    val fileName = urlString.split("/").last
+    val hiveTestJarsDir = new File("/tmp/test-spark/hiveTestJars")
+    if (!hiveTestJarsDir.exists()) {
+      hiveTestJarsDir.mkdirs()
+    }
+    val targetFile = new File(hiveTestJarsDir, fileName)
+    if (!targetFile.exists() || !(targetFile.length() > 0)) {
```

Comment:
<= 0? total nit.

Reply:
`!(targetFile.length() > 0)` avoids file corruption, because all tests use this file, we don't need it after changing the temporary directory to `Utils.createTempDir()`.